src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -221,6 +222,84 @@
         ),
         terminator=OneOf("ORDER", "LIMIT", "HAVING", "QUALIFY", "WINDOW"),
     ),
+    PatternQuantifierGrammar=Sequence(
+        OneOf(
+            StringParser(
+                "+",
+                SymbolSegment,
+                name="plus",
+                type="plus",
+            ),
+            StringParser(
+                "*",
+                SymbolSegment,
+                name="star",
+                type="star",
+            ),
+            StringParser(
+                "?",
+                SymbolSegment,
+                name="question_mark",
+                type="question_mark",
+            ),
+            Bracketed(
+                OneOf(
+                    Ref("NumericLiteralSegment"),
+                    Sequence(
+                        Ref("NumericLiteralSegment"),
+                        Ref("CommaSegment"),
+                    ),
+                    Sequence(
+                        Ref("CommaSegment"),
+                        Ref("NumericLiteralSegment"),
+                    ),
+                    Sequence(
+                        Ref("NumericLiteralSegment"),
+                        Ref("CommaSegment"),
+                        Ref("NumericLiteralSegment"),
+                    ),
+                ),
+                bracket_type="curly",
+                bracket_pairs_set="bracket_pairs",
+            ),
+        ),
+        # To put a quantifier into “reluctant mode”
+        StringParser(
+            "?",
+            SymbolSegment,
+            name="question_mark",
+            type="question_mark",
+            optional=True,
+        ),
```

Comment:
Again why not use ` `Ref("QuestionMarkSegment")`? Especially as using this twice now.

Reply:
Wasn't aware of this one from bigquery! will replace this thanks 😄